code/add_aws_regions_to_database.py

```diff
@@ -0,0 +1,119 @@
+import argparse
+import Querying_Database as qdb
+
+from collections import defaultdict
+from geopy.distance import geodesic
+from math import inf
+from shapely import wkt, shortest_line, LineString, Point
+
+def find_closest_paths(lat: float, lon: float, db_path: str, max_distance: float):
+    querier = qdb.queryDatabase(db_path)
+    query = f"""SELECT * FROM standard_paths;"""
+    results = querier.execute_query(query)
+    
+    paths = []
+    for row in results:
+        fc = row[0]
+        fs = row[1]
+        fcc = row[2]
+        tc = row[3]
+        ts = row[4]
+        tcc = row[5]
+        dist_km = float(row[6])
+        path_wkt = row[7]
+        edge = ((fc, fs, fcc), (tc, ts, tcc))
+
+
+        linestring = wkt.loads(path_wkt)
+        # The linestring is an existing edge in the graph, and the point is the
+        # location of the cloud region. We get the shortest line between these
+        # two geometries, which gives us the shortest distance from a cloud
+        # region to an existing edge in the graph.
+        ps = shortest_line(Point(lon, lat), linestring).coords
```

Comment:
Are we sure that shapely's shortest_line works properly on earth's sphere? I'm going through their [documentation](https://shapely.readthedocs.io/en/maint-2.0/index.html) and the package seems only suitable for perfect 2d Cartesian plane. I imagine this might give inaccurate results as we get closer to the poles.

Or maybe this works in a small enough region (<5km iirc) such that that difference is minimal for getting the tangent point, and you still use proper geodistance (`geodesic` below) for the actual distance?

Reply:
This is a good point.. I'm not sure that it does. Actually it looks like we may want to use a GeoPandas dataframe instead. It seems that we can give that a coordinate system and it handles the conversions: https://github.com/geopandas/geopandas/issues/3045

It may not matter for short distances but it's probably best to use it anyways.

Reply:
In fact, that might be why the points weren't lining up exactly when plotted.

Reply:
Okay @nilyibo , I reimplemented it using geopandas in the latest commit. Although I don't think it makes a large difference. The points were very slightly different. Let me know what you think, I think it'd be fine to keep the old method if it's more understandable, even if the new method is probably more correct.

And the reason the endpoint doesn't line up with the datacenter is because the datacenter location is taken from two different data sources (one is the iGDB database and one is ITDK).